src/test/java/com/dept/firstAssigmentDept/service/CatalogItemsServiceImpTest.java

```diff
@@ -0,0 +1,244 @@
+package com.dept.firstAssigmentDept.service;
+
+import com.dept.firstAssigmentDept.converter.CatalogItemDTOToCatalogItem;
+import com.dept.firstAssigmentDept.converter.CatalogItemToCatalogItemDTO;
+import com.dept.firstAssigmentDept.dto.CatalogItemDTO;
+import com.dept.firstAssigmentDept.exception.NotFoundException;
+import com.dept.firstAssigmentDept.model.CatalogItem;
+import com.dept.firstAssigmentDept.repository.CatalogItemsRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class CatalogItemsServiceImpTest {
+
+    @Mock
+    private CatalogItemsRepository catalogItemsRepository;
+    @InjectMocks
+    private CatalogItemsServiceImp catalogItemsService;
+    @Mock
+    private CatalogItemToCatalogItemDTO catalogItemToCatalogItemDTO;
+    @Mock
+    private CatalogItemDTOToCatalogItem catalogItemDTOToCatalogItem;
+
+    @Test
+    void getAllCatalogItems_shouldReturnCatalogItems() {
+
+        CatalogItem catalogItem1 = createCatalogItem(1L,"TestItem1","The new super TestItem", new BigDecimal(99.99));
+        CatalogItem catalogItem2 = createCatalogItem(2L,"TestItem2","The super TestItem2", new BigDecimal(19.99));
+        CatalogItem catalogItem3 = createCatalogItem(3L,"TestItem3","The super TestItem3", new BigDecimal(1999.99));
+
+        List<CatalogItem> catalogItems = new ArrayList<>(Arrays.asList(catalogItem1, catalogItem2, catalogItem3));
+
+        when(catalogItemsRepository.findAll()).thenReturn(catalogItems);
+
+        List<CatalogItemDTO> actualCatalogItems = catalogItemsService.getAllCatalogItems();
+
+        assertThat(actualCatalogItems, is(notNullValue()));
+        assertThat(actualCatalogItems.size(), equalTo(3));
```

Comment:
Since you have hamcrest you can try something like Hamcrest.containsInAnyOrder https://www.baeldung.com/java-assert-lists-equality-ignore-order#using-hamcrest. With this check you can remove these 2 since they would be redundunt

Reply:
Since you are returning DTO from the service then this would not work so you can ignore it